test/parallel/test-stream2-finish-pipe.js

```diff
@@ -35,7 +35,11 @@ w._write = function(data, encoding, cb) {
 
 r.pipe(w);
 
-// This might sound unrealistic, but it happens in net.js. When
-// `socket.allowHalfOpen === false`, EOF will cause `.destroySoon()` call which
-// ends the writable side of net.Socket.
-w.end();
+// end() must be called in nextTick or a WRITE_AFTER_END error occurs.
```

Comment:
Have we got a test that cover this case?

Reply:
Yes, I added a test in this PR, test-stream2-finish-pipe-error.js